### DISCUSSION TITLE
Can a device support multiple profiles?

### DISCUSSION THREAD

**Discussion by dmocek** [edgexfoundry/community#46] [General]
Can a device support multiple profiles?

**Comment by lenny-goodell** [comment]
Device Profiles? Yes. 
Example here: https://github.com/edgexfoundry/device-rest-go/tree/main/cmd/res/profiles

All profiles in this folder are used.

**Comment by dmocek** [comment]
Thanks Lenny.  Jim White said yes, as long as all profiles are using the same protocol.

**Comment by dmocek** [comment]
How do you specify multiple Device Profiles in a Device?  profileName in Device is a string: https://app.swaggerhub.com/apis/storysu2/edgex-foundry_core_metadata_api/2.3.0#/default/post_device

**Reply by cloudxxx8** [reply]
I don't think there is way to do that.  The workaround is to combine multiple Device Profiles into one profile.  You can do that manually or try the online tool https://dct.iotechsys.com/

**Comment by dmocek** [comment]
Understand, thanks.
